### PR TITLE
[EDX-424] Create new channels section and migrate content into it

### PR DESCRIPTION
**Pull request by m-hulbert**
## Description

This PR creates a new `Pub/Sub channels` section and migrates content into the following pages:

* Channels
* Messages
* Channel options
  * Rewind
  * Deltas
  * Encryption

The commits are structured as follows:

* Comment out migrated content from old pages
* Add migrated (and new) content into new section
* Update the table of contents
* Update links from migrated sections

See the following JIRAs for content that has been migrated: [EDX-424](https://ably.atlassian.net/browse/EDX-424), [EDX-425](https://ably.atlassian.net/browse/EDX-425), [EDX-426](https://ably.atlassian.net/browse/EDX-426), [EDX-427](https://ably.atlassian.net/browse/EDX-427), [EDX-428](https://ably.atlassian.net/browse/EDX-428) & [EDX-429](https://ably.atlassian.net/browse/EDX-429).

## Review

Ensure the content in the new `/channels/` section is coherent and it covers all content that has been commented out.

### CODE REVIEW

File: content/channels/options/rewind.textile [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+
+The @rewind@ channel option enables a client to specify where to start an attachment from, when attaching to a channel. This can be a point in time in the past, or a given number of messages. When the client attaches, they will receive the previously published messages. 
+
+As @rewind@ only applies to channel subscriptions, it is only available when using the realtime interface of an Ably SDK, or when using "SSE":LINKME or "MQTT":LINKME.
+
+Rewind only has an effect on an initial channel attachment. Any subsequent reattachment of the same channel on the same connection, in order to resume the connection, will attempt to resume with continuity from the point at which the connection dropped. There are a few exceptions to this, such as clients using "@recover@ mode":LINKME, where the previous attachment state is not preserved.
+
+h2(#subscribe). Subscribe with rewind
+
+The @rewind@ property of @params@ can be set a point in time in the past, or a given number of messages.
```
Comment (caitlindavey) [inline content/channels/options/rewind.textile:23]: "at a point in time in the past or for a given number of messages."

File: content/channels/options/rewind.textile [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+  var channelOptions = new ChannelOptions();
+  channelOptions.Params = channelParams;
+  var channel = ably.Channels.Get("{{RANDOM_CHANNEL_NAME}}", channelOptions);
+
+  channel.Subscribe(message => {
+      Console.WriteLine(message.Data.ToString());
+  });
+```
+
+h3(#number). Number of messages
```
Comment (caitlindavey) [inline content/channels/options/rewind.textile:91]: Can we make this heading more clear? Like "Retrieving past messages"

File: content/channels/options/rewind.textile [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+
+```[swift]
+let realtime = ARTRealtime(key: "{{API_KEY}}")
+let channel = realtime.channels.get("[?rewind=1]{{RANDOM_CHANNEL_NAME}}")
+channel.subscribe { message in
+    print("Received: \(message.data)")
+}
+```
+
+h3(#time). Period of time
```
Comment (caitlindavey) [inline content/channels/options/rewind.textile:157]: Same with this heading.

Comment (caitlindavey) [reply content/channels/options/rewind.textile:157]: E.g. "Retrieving messages from a period of time"

File: content/channels/options/index.textile [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+  - javascript
+  - nodejs
+  - objc
+  - php
+  - python
+  - ruby
+  - swift
+---
+
+Channel options can be used to customize the functionality of channels. This includes enabling features such as "encryption":/channels/options/encryption and "deltas":/channels/options/deltas, or for a client to retrieve messages published prior to it attaching to a channel using "rewind":/channels/options/rewind.
```
Comment (caitlindavey) [inline content/channels/options/index.textile:17]: Something that is missing for me from this doc as a whole is WHY/WHEN you'd want to rewind, especially by like one message or 5 seconds. Putting in some recommendations might be helpful guidance.

File: content/channels/index.textile [hunk 5]
```diff
@@ -0,0 +1,1240 @@
...
+        "code": 40160
+      }
+    }
+  ]
+}
+```
+
+h3(#transient-publish). Transient publish
+
+Transient publishing is when a client publishes messages without attaching to a channel. This is a feature of the realtime interface of "certain Ably SDKs":/LINKME. Transient publishing can be beneficial if you intend to publish to many channels as it removes the need to attach to a channel each time you publish. It also avoids a client subscribing to messages which avoids messages being sent to it redundantly.
```
[the remark this answers is not in the document]
Comment (m-hulbert) [reply content/channels/index.textile:541]: The placeholder link here will link out to the feature matrix we're implementing that should cover this.

File: content/channels/index.textile [hunk 6]
```diff
@@ -0,0 +1,1240 @@
...
+        System.out.println("Attach failed: " + reason.message);
+      }
+    }
+  }
+});
+channel.attach();
+```
+
+```[csharp]
+IRealtimeChannel privateChannel = realtime.Channels.Get("private:chatroom");
```
[the remark this answers is not in the document]
Comment (m-hulbert) [reply content/channels/index.textile:1194]: Yes, this is correct. Worth checking though 🙂

File: content/channels/index.textile [hunk 7]
```diff
@@ -0,0 +1,1240 @@
...
+
+```[flutter] 
+final channel = rest.channels.get('channelName');
+```
+
+h2(#attach). Attach to a channel
+
+Attaching to a channel ensures that it is created in the Ably system and that all messages published on the channel are received by any channel listeners registered by calling "subscribe()":/api/realtime-sdk/channels#subscribe. Ably will start to stream messages to a client as soon as they have attached, regardless of whether or not they have yet subscribed. Attach is only available to the realtime interface.
+
+Channels are not pre-configured or provisioned by Ably in advance. They are created on demand when clients attach, and remain active until such time that there are no remaining attached clients.
```
[the remark this answers is not in the document]
Comment (m-hulbert) [reply content/channels/index.textile:128]: This is covered in the section that follows regarding "detaching". Do you think we should cover it here too?

**Comment by caitlindavey** [comment]
Reviewed! I'll let you resolve the conflicts prior to merging.

**Review by caitlindavey — APPROVED** [review]

**Pull request merged**